src/proxy/offline/proxy.js

```diff
@@ -178,30 +178,43 @@ export default _.defaults(
                         result.testContext = {
                             state: states.testSession.closed
                         };
-                        if (isOffline) {
-                            return new Promise(function() {
-                                offlineSyncModal(self)
-                                    .on('proceed', function() {
-                                        self.syncData()
-                                            .then(function() {
+                        const offlineSync = function() {
+                            offlineSyncModal(self)
+                                .on('proceed', function() {
+                                    self.syncData()
+                                        .then(function() {
+                                            // if is online resolve promise 
+                                            if (self.isOnline()) {
                                                 return resolve(result);
-                                            })
-                                            .catch(function() {
-                                                return resolve({ success: false });
-                                            });
-                                    })
-                                    .on('secondaryaction', function() {
-                                        self.initiateDownload().catch(function() {
+                                            }
+                                        })
+                                        .catch(function() {
                                             return resolve({ success: false });
                                         });
+                                    })
+                                .on('secondaryaction', function() {
+                                    self.initiateDownload().catch(function() {
+                                        return resolve({ success: false });
                                     });
-                            }).catch(function() {
-                                return resolve({ success: false });
-                            });
+                                });
+                            };
+                        if (isOffline) {
+                            return new Promise(offlineSync)
```

Comment:
This is useless since the `return` is inside a promise constructor. It won't do anything, the promise won't be consumed. `return offlineSync()` will do the same